src/entity.ts

```diff
@@ -882,15 +883,17 @@ export namespace entity {
           }
         });
       } else if (firstPathPartIsArray && hasWildCard && remainderPath === '*') {
-        const array = entity.properties![firstPathPart].arrayValue;
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        array.values.forEach((value: any) => {
-          if (value.entityValue) {
-            excludePathFromEntity(value.entityValue, '.*');
-          } else {
-            excludePathFromEntity(value, '');
-          }
-        });
+        if (entity.properties![firstPathPart]) {
```

Comment:
So we just check to see that this property is defined before we move further as that prevents the error from being thrown. Makes sense. There seems to be no value in further traversing a path that the entity will not have any corresponding properties for.